pkg/certificate/providers/config.go

```diff
@@ -158,6 +169,21 @@ func (c *MRCProviderGenerator) getHashiVaultOSMCertificateManager(mrc *v1alpha2.
 	return vaultClient, id, nil
 }
 
+// getHashiVaultOSMToken returns the Hashi Vault token
+func getHashiVaultOSMToken(secretKeyRef *v1alpha2.SecretKeyReferenceSpec, kubeClient kubernetes.Interface) (string, error) {
+	tokenSecret, err := kubeClient.CoreV1().Secrets(secretKeyRef.Namespace).Get(context.TODO(), secretKeyRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error retrieving Hashi Vault token secret %s/%s: %w", secretKeyRef.Namespace, secretKeyRef.Name, err)
+	}
+
+	token, ok := tokenSecret.Data[secretKeyRef.Key]
+	if !ok {
+		return "", fmt.Errorf("failed to get Hashi Vault token with key %s from secret %s/%s", secretKeyRef.Key, secretKeyRef.Namespace, secretKeyRef.Name)
```

Comment:
Maybe something closer to "key not found" would be a bit more explicit and more obvious how to fix. Just from this message it's not super clear where the problem is exactly.